src/ode_solver/low_storage_runge_kutta_ode_solver.cpp

```diff
@@ -0,0 +1,270 @@
+    dt = pow(epsilon[0], 1.0 * beta1/rk_order) * pow(epsilon[1], 1.0 * beta2/rk_order) * pow(epsilon[2], 1.0 * beta3/rk_order) * dt;
+    return dt;
+}
+
+template <int dim, typename real, int n_rk_stages, typename MeshType> 
+double LowStorageRungeKuttaODESolver<dim,real,n_rk_stages, MeshType>::get_automatic_initial_step_size (real dt, const bool /*pseudotime*/)
+{
+    // h will store the starting step size
+    double h0 = 0.0;
+    double h1 = 0.0;
+
+    // d estimates the derivate of the solution
+    double d0 = 0.0;
+    double d1 = 0.0;
+    double d2 = 0.0;
+
+    /// Storage of the solution to calculate the initial time step
+    dealii::LinearAlgebra::distributed::Vector<double> u_n;
+    dealii::LinearAlgebra::distributed::Vector<double> rhs_initial;
+
+    this->solution_update = this->dg->solution;
+    u_n.reinit(this->solution_update);
+    rhs_initial.reinit(this->solution_update);
+    storage_register_1.reinit(this->solution_update);
+    storage_register_1 = this->dg->solution;
+    u_n = storage_register_1;
+    rhs_initial = storage_register_1;
+
+    d0 = u_n.linfty_norm();
+    
+    this->dg->solution = rhs_initial;
+    this->dg->assemble_residual();
+    this->dg->apply_inverse_global_mass_matrix(this->dg->right_hand_side, rhs_initial);
+
+    d1 = rhs_initial.linfty_norm();
+
+    if (d0 < pow(10, -5) || d1 < pow(10, -5)){
+        h0 = pow(10, -6);
+    }else{
+        h0 = 0.01 * d0 / d1;
+    }
+
+    u_n.add(h0, rhs_initial);
+    this->dg->solution = u_n;
+    this->dg->assemble_residual();
+    this->dg->apply_inverse_global_mass_matrix(this->dg->right_hand_side, u_n);
+    
+    // Calculate d2
+    u_n.add(-1, rhs_initial);
+    d2 = u_n.linfty_norm();
+    d2 /= h0;
+
+    if (std::max(d1, d2) <= pow(10, -15))
+    {
+        h1 = std::max(pow(10, -6), pow(10, -3));
+    }
+    else
+    {
+        h1 = 0.01 / std::max(d1, d2);
+    }
+
+    dt = std::min(100 * h0, h1);
+    this->dg->solution = storage_register_1;
+    return dt;
+}
+
+
+template <int dim, typename real, int n_rk_stages, typename MeshType> 
+void LowStorageRungeKuttaODESolver<dim,real,n_rk_stages, MeshType>::allocate_ode_system ()
+{
+    this->solution_update.reinit(this->dg->solution);
+    storage_register_1.reinit(this->solution_update);
+
+    atol = this->ode_param.atol;
+    rtol = this->ode_param.rtol;
+    rk_order = this->ode_param.rk_order;
+    is_3Sstarplus = this->ode_param.is_3Sstarplus;
+    num_delta = this->ode_param.num_delta;
+    beta1 = this->ode_param.beta1;
+    beta2 = this->ode_param.beta2;
+    beta3 = this->ode_param.beta3;
+
+    global_size = dealii::Utilities::MPI::sum(storage_register_1.local_size(), this->mpi_communicator);
+    this->pcout << "Allocating ODE system..." << std::flush;
+    this->solution_update.reinit(this->dg->right_hand_side);
+    if(this->all_parameters->use_inverse_mass_on_the_fly == false) {
+        this->pcout << " evaluating inverse mass matrix..." << std::flush;
```

Comment:
I think the code still works if use_inverse_mass_on_the_fly == false because I added an abort statement and tried to run the 1D linear advection case and it aborts, however it usually runs

Reply:
Thanks! (Note for documentation purposes: we agreed to keep the next few lines commented as they'll be needed for an RRK implementation!)